fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Deserializable.kt

```diff
@@ -1,16 +1,32 @@
 package com.github.kittinunf.fuel.core
 
-import com.github.kittinunf.fuel.core.requests.AsyncTaskRequest
-import com.github.kittinunf.fuel.core.requests.SuspendingRequest
+import com.github.kittinunf.fuel.core.requests.CancellableRequest
+import com.github.kittinunf.fuel.core.requests.RequestTaskCallbacks
+import com.github.kittinunf.fuel.core.requests.suspendable
+import com.github.kittinunf.fuel.core.requests.toTask
 import com.github.kittinunf.result.Result
+import com.github.kittinunf.result.getOrElse
 import com.github.kittinunf.result.map
 import com.github.kittinunf.result.mapError
 import java.io.InputStream
 import java.io.Reader
 
-typealias HandlerWithResult<T> = (Request, Response, Result<T, FuelError>) -> Unit
-
+/**
+ * Generic interface for [Response] deserialization.
+ *
+ * @note you are responsible of using the [Response] [Body] [InputStream] and closing it when you're done. Failing to do
+ *   so can result in hanging connections if used in conjunction with [com.github.kittinunf.fuel.toolbox.HttpClient].
```

Comment:
is this a good idea to hand this responsibility to users of the lib? would we be able to make an annotation to check that close is being called?

Reply:
That would be sweet.

Currently this is already the case, I am just documenting it haha.